specs/interop/predeploys.md

```diff
@@ -214,15 +232,15 @@ function relayMessage(uint256 _destination, uint256 _source, uint256 _nonce, add
       tstore(CROSS_DOMAIN_MESSAGE_SOURCE_SLOT, _source)
     }
 
+    sentMessages[messageHash] = true;
```

Comment:
should we keep this before the call? it's not stored in transient storage, so we don't use reentrancy protection

Reply:
This is a simple example and doesn't necessarily need to be the full implementation but instead is meant to illustrate how the system works. Perhaps making it be python would make that more clear.

This follows the [checks, effects, interactions pattern](https://docs.soliditylang.org/en/v0.6.11/security-considerations.html#use-the-checks-effects-interactions-pattern) which is generally considered best practice but has been supplemented with [FREI-PI](https://www.nascent.xyz/idea/youre-writing-require-statements-wrong)